src/KubernetesClient/WatcherDelegatingHandler.cs

```diff
@@ -64,6 +66,37 @@ protected override bool TryComputeLength(out long length)
                 length = 0;
                 return false;
             }
+        }
+        internal class PeekableStreamReader : StreamReader
```

Comment:
I'm worried that if someone calls any of the other `ReadBlock`, `Read` etc methods this is going to break...

At the very least let's override those messages and have them throw an exception so that we detect that it's happening...

Reply:
Makes sense to me. I have updated the pull request to override all the remaining `Read` methods to throw `NotImplementedException`

Reply:
@brendanburns The build is failing but I have no clue why and I don't have permission to re-trigger the build. Would you please point me to what should I check?

Reply:
@brendanburns, Was wondering if you have time to review the updated request